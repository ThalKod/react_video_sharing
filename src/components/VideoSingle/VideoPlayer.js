import React from "react";
import "video-react/dist/video-react.css";
import {
  Player,
  BigPlayButton,
  LoadingSpinner,
  ControlBar,
  ReplayControl,
  ForwardControl,
  PlaybackRateMenuButton,
  VolumeMenuButton } from "video-react";

export default function({ videoUrl }){
  return(
      <div className="sv-video">
        <Player
          src={videoUrl}
        >
          <BigPlayButton position="center" />
          <LoadingSpinner />
          <ControlBar>
            <PlaybackRateMenuButton rates={[5, 2, 1, 0.5, 0.1]} order={2.1} />
            <ReplayControl seconds={10} order={2.2}/>
            <ForwardControl seconds={10} order={3.2} />
            <VolumeMenuButton vertical/>
          </ControlBar>
        </Player>
      </div>
  )
}
