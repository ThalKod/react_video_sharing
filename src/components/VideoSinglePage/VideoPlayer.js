import React from "react";
import "video-react/dist/video-react.css";
import { Player, BigPlayButton } from "video-react";

export default function(){
  return(
      <div className="sv-video">
        <Player
          src="https://cdn.filestackcontent.com/Cd1mB8L8Tt21TmyMSgav"
        >
          <BigPlayButton position="center" />
        </Player>
      </div>
  )
}
