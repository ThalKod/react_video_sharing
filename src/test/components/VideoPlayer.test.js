import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import VideoPlayer from "components/VideoSingle/VideoPlayer";

describe("VideoPlayer",() => {
  it("should render the video player correctly", () => {
    const videoUrl = "myurl";
    const wrapper = shallow(<VideoPlayer videoUrl={videoUrl}/>);

    expect(toJson(wrapper)).toMatchSnapshot();
  })
});
