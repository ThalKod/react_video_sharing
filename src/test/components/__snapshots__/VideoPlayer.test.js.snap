// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayer should render the video player correctly 1`] = `
<div
  className="sv-video"
>
  <Player
    aspectRatio="auto"
    autoplay={true}
    fluid={true}
    muted={false}
    playsInline={false}
    preload="auto"
    src="myurl"
  >
    <BigPlayButton
      position="center"
    />
    <LoadingSpinner />
    <ControlBar
      autoHide={true}
      disableCompletely={false}
    >
      <PlaybackRateMenuButton
        order={2.1}
        rates={
          Array [
            5,
            2,
            1,
            0.5,
            0.1,
          ]
        }
      />
      <ReplayControl
        order={2.2}
        seconds={10}
      />
      <ForwardControl
        order={3.2}
        seconds={10}
      />
      <VolumeMenuButton
        vertical={true}
      />
    </ControlBar>
  </Player>
</div>
`;
